### DISCUSSION TITLE
Change eval for Function class

### DISCUSSION THREAD

**Discussion by francisco-giancarelli-crombie** [NucleoidAI/Nucleoid#49] [Ideas]
I see in the code that nucleoid runtime is using eval to ultimately run the code. eval is expensive and will make the runtime far more slow than plain NodeJS, more on that here https://www.educative.io/answers/eval-vs-function-in-javascript. I think nucleoid could use Function class instead to get performance closer to NodeJS.

**Comment by canmingir** [comment]
Mostly we are using `eval` for assigning/resolving value of variable, in our case, it looks like `eval` is much faster than `Function`

```
let a;

const b1 = Date.now();
for (let i = 0; i < 1_000_000; i++) {
  a = Function("return 1 + 2")();
}
const a1 = Date.now();
console.log(a1 - b1); // ~375

const b2 = Date.now();
for (let i = 0; i < 1_000_000; i++) {
  a = eval("1 + 2");
}
const a2 = Date.now();
console.log(a2 - b2); // ~125
```

Do you have anything else in your mind?
